File: src/Wordki.Core/Domain/Result.cs
using Newtonsoft.Json;
using System;
using Wordki.Core.Enums;

namespace Wordki.Core
{
    public class Result
    {
        public long Id { get; set; }
        public long GroupId { get; set; }
        [JsonIgnore]
        public Group Group { get; set; }
        public long UserId { get; set; }
        public short Correct { get; set; }
        public short Accepted { get; set; }
        public short Wrong { get; set; }
        public short Invisible { get; set; }
        public short TimeCount { get; set; }
        public TranslationDirection TranslationDirection { get; set; }
        public LessonType LessonType { get; set; }
        public DateTime DateTime { get; set; }
        public int State { get; set; }
        public DateTime? LastChange { get; set; }

        public Result()
        {
            Id = -1;
            State = int.MaxValue;
        }

    }
}

File: client/src/app/common/models/Result.ts
import { TranslationDirectionEnum } from './TranslationDirectionEnum';
import { LessonTypeEnum } from './LessonTypeEnum';

export class Result {
  id: string;
  userId:string;
  groupId: string;
  correct = 0;
  accepted = 0;
  wrong = 0;
  invisible = 0;
  timeCount = 0;
  translationDirection = TranslationDirectionEnum.Default;
  lessonType = LessonTypeEnum.Fiszki;
  dateTime: Date = new Date();

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }
}
